nano-sync/src/circuits/mnt6/pk_tree_0.rs

```diff
@@ -0,0 +1,238 @@
+
+        let vk_child = VerifyingKey::deserialize(&mut file).unwrap();
+
+        // Allocate all the constants.
+        #[allow(unused_mut)]
+        let mut cost = start_cost_analysis!(cs, || "Alloc constants");
+
+        let vk_child_var = TheVkGadget::alloc_constant(cs.ns(|| "alloc vk child"), &vk_child)?;
+
+        // Allocate all the private inputs.
+        next_cost_analysis!(cs, cost, || { "Alloc private inputs" });
+
+        let left_proof_var =
+            TheProofGadget::alloc(cs.ns(|| "alloc left proof"), || Ok(&self.left_proof))?;
+
+        let right_proof_var =
+            TheProofGadget::alloc(cs.ns(|| "alloc right proof"), || Ok(&self.right_proof))?;
+
+        // Allocate all the public inputs.
+        next_cost_analysis!(cs, cost, || { "Alloc public inputs" });
+
+        let pk_commitment_var = UInt8::alloc_input_vec(
+            cs.ns(|| "alloc public keys commitment"),
+            self.pks_commitment.as_ref(),
+        )?;
+
+        let prepare_signer_bitmap_var = UInt8::alloc_input_vec(
+            cs.ns(|| "alloc prepare signer bitmap"),
+            self.prepare_signer_bitmap.as_ref(),
+        )?;
+
+        let left_prepare_agg_pk_commitment_var = UInt8::alloc_input_vec(
+            cs.ns(|| "alloc left prepare aggregate pk commitment"),
+            self.left_prepare_agg_pk_commitment.as_ref(),
+        )?;
+
+        let right_prepare_agg_pk_commitment_var = UInt8::alloc_input_vec(
+            cs.ns(|| "alloc right prepare aggregate pk commitment"),
+            self.right_prepare_agg_pk_commitment.as_ref(),
+        )?;
+
+        let commit_signer_bitmap_var = UInt8::alloc_input_vec(
+            cs.ns(|| "alloc commit signer bitmap"),
+            self.commit_signer_bitmap.as_ref(),
+        )?;
+
+        let left_commit_agg_pk_commitment_var = UInt8::alloc_input_vec(
+            cs.ns(|| "alloc left commit aggregate pk commitment"),
+            self.left_commit_agg_pk_commitment.as_ref(),
+        )?;
+
+        let right_commit_agg_pk_commitment_var = UInt8::alloc_input_vec(
+            cs.ns(|| "alloc right commit aggregate pk commitment"),
+            self.right_commit_agg_pk_commitment.as_ref(),
+        )?;
+
+        let position_var =
+            UInt8::alloc_input_vec(cs.ns(|| "alloc position"), self.position.as_ref())?
+                .pop()
+                .unwrap();
+
+        // Calculate the position for the left and right child nodes. Given the current position P,
+        // the left position L and the right position R are given as:
+        //    L = 2 * P
+        //    R = 2 * P + 1
+        // For efficiency reasons, we calculate the positions using bit manipulation.
+        next_cost_analysis!(cs, cost, || { "Calculate positions" });
+
+        let mut bits = position_var.into_bits_le();
+
+        bits.pop();
+        bits.insert(0, Boolean::Constant(false));
+        let left_position = vec![UInt8::from_bits_le(&bits)];
```

Comment:
I would add a comment to make clear what you're doing here (additionally to the high level description above).
```suggestion
        // Calculate `P << 1`.
        bits.pop();
        bits.insert(0, Boolean::Constant(false));
        let left_position = vec![UInt8::from_bits_le(&bits)];
```

Reply:
Done.